logging/google/cloud/logging/_gax.py

```diff
@@ -603,8 +603,11 @@ def _log_entry_mapping_to_pb(mapping):
             entry_pb.labels[key] = value
 
     if 'jsonPayload' in mapping:
-        for key, value in mapping['jsonPayload'].items():
-            entry_pb.json_payload[key] = value
+        # NOTE: ``json.dumps`` is wasteful here because internally,
+        #       ``Parse`` will just call ``json.loads``. However,
+        #       there is no equivalent public function to parse on raw
+        #       dictionaries, so we waste cycles on parse/unparse.
```

Comment:
@tseaver My editorializing here was based on a vague-ish memory I have of you struggling with the lack of a public function for this. I can remove the comment if I am remembering incorrectly.